src/formatting_stack/core.clj

```diff
@@ -47,30 +48,51 @@
                              :level     :exception}])))))
            (doall)))))
 
-(defn format! [& {:keys [strategies
-                         third-party-indent-specs
-                         formatters
-                         linters
-                         processors
-                         reporter
-                         in-background?]}]
+(speced/defn format! [& {:keys                                             [^vector? strategies
```

Comment:
Not a big fan of the factory-concept here. We could assoc the strategies in `format!` (they are impl detail anyway), which would remove the need of creating the formatters/etc here, retaining the vectors as params. 

`third-party-indent-specs` is still an issue, suggestion posted below

Reply:
> We could assoc the strategies in format! (they are impl detail anyway)

See my comment beginning with `The first thing that comes to mind [...]`